### PR TITLE
Fixed bug in determine_tier_program

### LINKED ISSUE #1228: Exception occurs when user enters an income that's too low in the financial aid calculator
2016-10-04T21:49:02.160992+00:00 app[web.1]: Traceback (most recent call last):
    2016-10-04T21:49:02.160993+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/core/handlers/base.py", line 149, in get_response
    2016-10-04T21:49:02.160994+00:00 app[web.1]:     response = self.process_exception_by_middleware(e, request)
    2016-10-04T21:49:02.160996+00:00 app[web.1]:     response = wrapped_callback(request, *callback_args, **callback_kwargs)
    2016-10-04T21:49:02.160995+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/core/handlers/base.py", line 147, in get_response
    2016-10-04T21:49:02.160997+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/newrelic-2.54.0.41/newrelic/hooks/framework_django.py", line 499, in wrapper
    2016-10-04T21:49:02.160998+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    2016-10-04T21:49:02.160997+00:00 app[web.1]:     return wrapped(*args, **kwargs)
    2016-10-04T21:49:02.160999+00:00 app[web.1]:     return view_func(*args, **kwargs)
    2016-10-04T21:49:02.160999+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/views/generic/base.py", line 68, in view
    2016-10-04T21:49:02.161000+00:00 app[web.1]:     return self.dispatch(request, *args, **kwargs)
    2016-10-04T21:49:02.161001+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/newrelic-2.54.0.41/newrelic/hooks/component_djangorestframework.py", line 27, in _nr_wrapper_APIView_dispatch_
    2016-10-04T21:49:02.161002+00:00 app[web.1]:     return wrapped(*args, **kwargs)
    2016-10-04T21:49:02.161003+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/rest_framework/views.py", line 466, in dispatch
    2016-10-04T21:49:02.161003+00:00 app[web.1]:     response = self.handle_exception(exc)
    2016-10-04T21:49:02.161004+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/rest_framework/views.py", line 463, in dispatch
    2016-10-04T21:49:02.161004+00:00 app[web.1]:     response = handler(request, *args, **kwargs)
    2016-10-04T21:49:02.161005+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/rest_framework/generics.py", line 192, in post
    2016-10-04T21:49:02.161006+00:00 app[web.1]:     return self.create(request, *args, **kwargs)
    2016-10-04T21:49:02.161006+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/rest_framework/mixins.py", line 21, in create
    2016-10-04T21:49:02.161007+00:00 app[web.1]:     self.perform_create(serializer)
    2016-10-04T21:49:02.161007+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/rest_framework/mixins.py", line 26, in perform_create
    2016-10-04T21:49:02.161008+00:00 app[web.1]:     serializer.save()
    2016-10-04T21:49:02.161009+00:00 app[web.1]:   File "./financialaid/serializers.py", line 76, in save
    2016-10-04T21:49:02.161010+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/db/models/manager.py", line 122, in manager_method
    2016-10-04T21:49:02.161009+00:00 app[web.1]:     date_exchange_rate=datetime.datetime.now()
    2016-10-04T21:49:02.161010+00:00 app[web.1]:     return getattr(self.get_queryset(), name)(*args, **kwargs)
    2016-10-04T21:49:02.161011+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/db/models/query.py", line 399, in create
    2016-10-04T21:49:02.161012+00:00 app[web.1]:     obj = self.model(**kwargs)
    2016-10-04T21:49:02.161012+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/db/models/base.py", line 439, in __init__
    2016-10-04T21:49:02.161013+00:00 app[web.1]:     setattr(self, field.name, rel_obj)
    2016-10-04T21:49:02.161014+00:00 app[web.1]:     (instance._meta.object_name, self.field.name)
    2016-10-04T21:49:02.161013+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.5/site-packages/django/db/models/fields/related_descriptors.py", line 199, in __set__
    2016-10-04T21:49:02.161015+00:00 app[web.1]: ValueError: Cannot assign None: "FinancialAid.tier_program" does not allow null values.

### PR DESCRIPTION
**Pull request by alvinsiu**
#### What are the relevant tickets?
Fixes #1228 

#### What's this PR do?
Fixed the bug where Exception occurs if no $0-income-threshold TierProgram has been created and an income lower than the lowest configured TierProgram is submitted.  It now raises `ImproperlyConfigured` if this case is encountered.

### CODE REVIEW

**Comment by noisecapella** [comment]
Functionality looks good

File: financialaid/api.py [hunk 1]
```diff
@@ -30,6 +30,8 @@ def determine_tier_program(program, income):
     # be the tier assigned to the user.
     tier_programs_set = program.tier_programs.filter(current=True, income_threshold__lte=income)
     tier_program = tier_programs_set.order_by("-income_threshold").first()
+    if tier_program is None:
+        raise ImproperlyConfigured("The $0-income-threshold TierProgram has not yet been configured for this Program.")
```
Comment (alvinsiu) [inline financialaid/api.py]: Yup!

**Comment by alvinsiu** [comment]
Merp - my added test got bundled with the merge commit because I hiccuped with git.  Might want to scan quickly through the 10 lines of changes?

https://github.com/mitodl/micromasters/pull/1256/files

**Comment by noisecapella** [comment]
Looks good to me :+1:

**Pull request merged**